nexus/reconfigurator/planning/src/planner.rs

```diff
@@ -2099,6 +2310,53 @@ impl<'a> Planner<'a> {
                     false
                 }
             }
+            ZoneKind::Nexus => {
+                // Get the nexus_generation of the zone being considered for shutdown
+                let zone_nexus_generation = match &zone.zone_type {
+                    BlueprintZoneType::Nexus(nexus_zone) => {
+                        nexus_zone.nexus_generation
+                    }
+                    _ => unreachable!("zone kind is Nexus but type is not"),
+                };
+
+                // Get the generation of the currently-executing Nexus zones.
+                //
+                // This presumably includes the currently-executing Nexus where
+                // this logic is being considered.
+                let Some(current_gen) = self.lookup_current_nexus_generation()
+                else {
+                    // If we don't know the current Nexus zone ID, or its
+                    // generation, we can't perform the handoff safety check.
+                    report.unsafe_zone(
+                        zone,
+                        Nexus {
+                            zone_generation: zone_nexus_generation,
+                            current_nexus_generation: None,
+                        },
+                    );
+                    return false;
+                };
```

Comment:
I feel like this case should be impossible now.  That would imply there was literally no Nexus zone in the blueprint at the current generation?